internal/controller/pod_controller.go

```diff
@@ -168,3 +189,136 @@ func (a *PodAdmissionWebhook) listOwners(ctx context.Context, object client.Obje
 	}
 	return owners, nil
 }
+
+// podEventHandlerRunner starts a podEventHandler when the manager
+// begins, ensuring that it only runs on the leader operator pod.
+type podEventHandlerRunner struct {
+	mgr manager.Manager
+	u   *workload.Updater
+	l   logr.Logger
+}
+
+// newPodEventHandlerRunner creates a new runner which will instantiate a
+// PodEventHandler during controller-runtime manager startup.
+func newPodEventHandlerRunner(mgr manager.Manager, u *workload.Updater, l logr.Logger) *podEventHandlerRunner {
+	return &podEventHandlerRunner{
+		mgr: mgr,
+		u:   u,
+		l:   l,
+	}
+}
+
+// newPodEventHandler creates a new PodEventHandler which implements
+func newPodEventHandler(ctx context.Context, c client.Client, u *workload.Updater, l logr.Logger) *podEventHandler {
+	return &podEventHandler{
+		ctx: ctx,
+		c:   c,
+		u:   u,
+		l:   l,
+	}
+}
+
+// NeedLeaderElection implements manager.LeaderElectionRunnable so that
+// the podEventHandler only runs on the leader, not on other redundant
+// operator pods.
+func (r *podEventHandlerRunner) NeedLeaderElection() bool {
+	return true
+}
+
+// Start implements manager.Runnable which will start the informer receiving
+// pod change events on the operator's leader instance.
+func (r *podEventHandlerRunner) Start(ctx context.Context) error {
+	h := newPodEventHandler(ctx, r.mgr.GetClient(), r.u, r.l)
+
+	i, err := r.mgr.GetCache().GetInformerForKind(ctx, schema.GroupVersionKind{
```

Comment:
Discussed Offline:
1. Filtering: This gets tricky, as filtering predicates would need to be added per AuthProxyWorkload. The workload resources (Deployment, StatefulSet) would need to be traversed to determine their pod selector, which could be used as a predicate. This could be done using a controler Watch() function, but it isn't clear if controller-runtime is architected to scale the number of watch events per pod selector. This also increases the complexity of the operator logic, and without a clear customer impacting issue, it isn't worth optimizing yet.
2. Cache Initialization: From empirical testing, the handler re-processes events of existing resources when the controller starts up.